x-pack/plugins/secops/public/components/timeline/body/renderers/auditd/translations.ts

```diff
@@ -0,0 +1,693 @@
+/*
```

Comment:
I am wondering if it will be better here to have the full sentence with variables so it is easier to translate it. I feel like they won't be able to get the job done and the meaning right if it is divided like it is.

Reply:
I can put examples of strings at the top so they get the gist of what is going on. I doubt I am going to duplicate the same strings ~178 times though since the way React is encouraging re-use is by splitting things up based around tokens.

Plus, it kind of is as written which is split up tokens because part of each sentence is data driven such as:

```
Session ${sessionId} ${secondary} as ${primary} @ {$hostName} logged in via ${processExecutable} with a ${result}
```

So some of it is English non-translatable as I can't control the i18n of auditd or beats on the backend returning `success` vs `failure` or the hostName or the user name or the executable name (which is typically English only).

However, I can add a sample string for anyone looking at the file to give them a clue on how to best translate it. Once you mix in data driven English words from 3rd party tooling with non-English words, you are at "best effort" from what I can tell.

Reply:
I was more thinking something like that
```
export const GET_MY_MESSAGE = (obj: InterfaceMyMsg) => i18n.translate(
  'xpack.secops.timeline.body.pinning.pinnnedWithNotesTooltip',
  {
    defaultMessage: 'Session ${sessionId} ${secondary} as ${primary} @ {$hostName} logged in via ${processExecutable} with a ${result}',
    values: {
      sessionId: obj.sessionId,
      secondary: obj.secondary,
      primary: obj.primary,
      etc....,
  }
);
```

Like that, I think translator can re-arrange words without too many issues. Maybe I am completely wrong but this feel more friendly and a better way to trasnlate when you have sentences.

Reply:
I think that if we are using component inside of the sentence we should go with `<FormattedMessage>` component.

Reply:
I wouldn't mind getting a feedback from the translators or the service/people we are using so I can understand how much work I should be doing vs. how much work I am making for them?

Otherwise, I would just imagine I am rearranging code in a _lot_ of areas which takes a lot of time to be possibly told to do it another way or it was readable either way? 

For now, if they are reading code to get context I can put comments above the code to let anyone know.

Reply:
Also, my example is not a simple dialog string with something like this:

```
defaultMessage: 'Session ${sessionId} ${secondary} as ${primary} @ {$hostName} logged in via ${processExecutable} with a ${result}',
```

That would be easier and I could do that and then take a an hour or two to refactor. What I have instead is a collection of drag and droppable components within other reusable components where it cascades down to construct rich inner reusable components. None of those reusable components is showing a simple one line string. It's instead a set of reusable ECS components in which even the variables can vary over time.

For example, sometimes you have ${processExecutable} and other times you don't. You have ${processName} but not ${processExecutable} however you want to display similarly.

There are a set of complex business rules just around `primary` and `secondary` and `user.name` to determine which string to produce.

I wish the problem set was as simple as static strings that do not have non-domain business rule dependent but rather a single set of variables. However, that's not what we have at the moment.

Reply:
It looks like we can pass components to `FormattedMessage` via `values`, which would in-turn allow new translations:

  ```js
  <FormattedMessage
    id="kbn.management.createIndexPattern.step.status.successLabel.successDetail"
    defaultMessage="{strongSuccess} Your index pattern matches {strongIndices}."
    values={{
      strongSuccess: (
        <strong>
          <FormattedMessage
            id="kbn.management.createIndexPattern.step.status.successLabel.strongSuccessLabel"
            defaultMessage="Success!"
          />
        </strong>),
      strongIndices: (
        <strong>
          <FormattedMessage
            id="kbn.management.createIndexPattern.step.status.successLabel.strongIndicesLabel"
            defaultMessage="{indicesLength, plural, one {# index} other {# indices}}"
            values={{ indicesLength: exactMatchedIndices.length }}
          />
        </strong>)
    }}
  />
  ```

Perhaps we can create a github issue to track this, and move forward with the PR as-is (with a link to the issue in this PR)?

Reply:
We have one PR open to track the final cleanup of translations and final sweep through the code. I would expect to get any feedback on the keys and do more of it towards the end or post beta:
https://github.com/elastic/ingest-dev/issues/232

FWIW, you don't need to use `<FormattedMessage>` to utilize `values` or `plurals`. You can still use the  values within your code through the same `i18n` interface and I think we are ok still using the same exact patterns we have been using which is the `translations` file.

For example I have these:

```ts
export const ROWS_5 = i18n.translate('xpack.secops.authenticationsTable.rows', {
  values: { numRows: 5 },
  defaultMessage: '{numRows} {numRows, plural, =0 {rows} =1 {row} other {rows}}',
});
```

And I use them like so:

```ts
const rowItems: ItemsPerRow[] = [
  {
    text: i18n.ROWS_5,
    numberOfRow: 5,
  },
```

To me that is more reusable if/when I need it because if I want to bubble that key of ROWS_5 with values and defaultMessage to a common location to be used it's much more simple than if I have to refactor and change around HTML chunks of code.

That's kind of my hesitation of using ` <FormattedMessage` is the extra chunky-ness of reading i18n code if I can get away without using it and now having to keep track of two different ways of doing things. 

You can do a search across all of Kibana using the string of `i18n.translate` and see plenty of clean examples using values and plurals and to me it looks so much simple to keep it in JS code rather than mixing between HTML and JS and having two ways of doing something if we can keep to one way of doing something:

Another example from our code base using values:
```ts
        formatYShort={(value: number) =>
          i18n.translate('xpack.apm.errorGroupDetails.occurrencesShortLabel', {
            defaultMessage: '{occCount} occ.',
            values: { occCount: value }
          })
        }
```

So at the moment, my opinion is to keep it to one way which is the JS way unless there is a feature that doesn't work using the `i18n.translate` technique.


If this helps, the script to output i18n to examine the keys and default values to a place such as `/tmp` would be this command to compare our keys to other project keys:

```ts
node scripts/i18n_extract.js --path x-pack/ --output-dir /tmp --output-format json5
```